File: restaurant-Services/src/main/java/com/foody/restaurantServices/domain/Restaurant.java
package com.foody.restaurantServices.domain;

import lombok.*;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;
@Document
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@ToString
public class Restaurant {
    @Id
    private String restaurantId;
    private String restaurantName;
    private String restaurantLocation;
    private String typeOfRestaurant;
    private String addressOfRestaurant;
    private  float restaurant_rating;
    private List<Cuisine> cuisineList;
    private byte[] picByte;
    private String fileName;
}

File: src/app/model/Restaurant.ts
import { Cuisine } from "./Cuisine";

export class Restaurant{
    
        restaurantId:any;
        restaurantName:any;
        restaurantLocation:any;
        typeOfRestaurant: any;
        addressOfRestaurant:any;
        picByte:any;
        restaurant_rating:any 
        cuisineList:Cuisine[] = []

constructor(       restaurantId:any,
        restaurantName:any,
        restaurantLocation:any,
    typeOfRestaurant: any,
    addressOfRestaurant:any,
    picByte:any,
    restaurant_rating:any
    ,cuisineList:Cuisine[]
     ){
           this.restaurantId= restaurantId;
            this.restaurantName= restaurantName;
             this.restaurantLocation= restaurantLocation;
              this.typeOfRestaurant= typeOfRestaurant;
              this.addressOfRestaurant= addressOfRestaurant;
              this.picByte= picByte;
              this.restaurant_rating= restaurant_rating;
              this.cuisineList = cuisineList;
}

}